include/seqan3/io/alignment_file/output.hpp

```diff
@@ -0,0 +1,940 @@
+                     detail::get_or_ignore<selected_field_ids::index_of(field::QUAL)>(t),
+                     detail::get_or_ignore<selected_field_ids::index_of(field::SEQ_QUAL)>(t),
+                     detail::get_or_ignore<selected_field_ids::index_of(field::MATE)>(t),
+                     detail::get_or_ignore<selected_field_ids::index_of(field::TAGS)>(t),
+                     detail::get_or_ignore<selected_field_ids::index_of(field::EVALUE)>(t),
+                     detail::get_or_ignore<selected_field_ids::index_of(field::BIT_SCORE)>(t));
+    }
+
+    /*!\brief            Write a record to the file by passing individual fields.
+     * \tparam arg_t     Type of the first field.
+     * \tparam arg_types Types of further fields.
+     * \param[in] arg    The first field to write.
+     * \param[in] args   Further fields.
+     *
+     * \details
+     *
+     * The fields are assumed to correspond to the field IDs given in selected_field_ids, however passing less
+     * is accepted if the format does not require all of them.
+     *
+     * ### Complexity
+     *
+     * Constant. TODO linear in the size of the written alignments?
+     *
+     * ### Exceptions
+     *
+     * Basic exception safety.
+     *
+     * ### Example
+     *
+     * ```cpp
+     * alignment_file_out fout{"/tmp/my.fasta"};
+     *
+     * auto it = fout.begin();
+     *
+     * for // ...
+     * {
+     *     std::string id;
+     *     dna5_vector seq;
+     *
+     *     // ...
+     *
+     *     fout.emplace_back(seq, id);
+     * }
+     * ```
+     */
+    template <typename arg_t, typename ... arg_types>
+    void emplace_back(arg_t && arg, arg_types && ... args)
+    {
+        push_back(std::tie(arg, args...));
+    }
+
+    /*!\brief            Write a range of records (or tuples) to the file.
+     * \tparam rng_t     Type of the range, must satisfy seqan3::output_range_concept and have a reference type that
+     *                   satisfies seqan3::tuple_like_concept.
+     * \param[in] range  The range to write.
+     *
+     * \details
+     *
+     * This function simply iterates over the argument and calls push_back() on each element.
+     *
+     * ### Complexity
+     *
+     * Linear in the number of records.
+     *
+     * ### Exceptions
+     *
+     * Basic exception safety.
+     *
+     * ### Example
+     *
+     * ```cpp
+     * alignment_file_out fout{"/tmp/my.fasta"};
+     *
+     * std::vector<std::tuple<dna5_vector, std::string>> range
+     * {
+     *     { "ACGT"_dna5, "First" },
+     *     { "NATA"_dna5, "2nd" },
+     *     { "GATA"_dna5, "Third" }
+     * }; // a range of "records"
+     *
+     * fout = range; // will iterate over the records and write them
+     * ```
+     */
+    template <typename rng_t>
+    alignment_file_out & operator=(rng_t && range)
+        requires std::ranges::InputRange<rng_t> && tuple_like_concept<reference_t<rng_t>>
```

Comment:
value_type_t and please guard for reference inside the rng_t.

Reply:
> value_type_t

why? dereferencing the iterator returns reference_t

> and please guard for reference inside the rng_t.

As noted before, our functions should work on references of ranges, it makes sense and simplifies the code.